docs/iceberg-catalog.md

```diff
@@ -0,0 +1,436 @@
+
+1. **List Warehouses**:
+   ```
+   datahub iceberg list
+   ```
+
+2. **Update Warehouse Configuration**:
+   ```
+   datahub iceberg update \
+     -w $WAREHOUSE_NAME \
+     -d $S3_DATA_ROOT \
+     -i $ICEBERG_CLIENT_ID \
+     --client_secret $ICEBERG_CLIENT_ID \
+     --region $S3_REGION \
+     --role $ICEBERG_ROLE
+   ```
+
+3. **Delete Warehouse**:
+   ```
+   datahub iceberg delete -w $WAREHOUSE_NAME
+   ```
+   Note: This command deletes Containers associated with Namespaces and Datasets associated with Tables and Views in this warehouse. However, it does not delete any backing data in the warehouse data_root location - it only deletes entries from the catalog.
+
+For additional options and help, run:
+```
+datahub iceberg [command] --help
+```
+
+### Migrating from Other Iceberg Catalogs
+
+When migrating from another Iceberg catalog, you can register existing Iceberg tables using the `system.register_table` command. This allows you to start managing existing Iceberg tables through DataHub without moving the underlying data.
+
+Example of registering an existing table:
+```
+# REGISTER EXISTING ICEBERG TABLE 
+call system.register_table('barTable', 's3://my-s3-bucket/my-data-root/fooNs/barTable/metadata/00000-f9dbba67-df4f-4742-9ba5-745aa2bb4076.metadata.json');
+select * from barTable;
+```
+
+### Security and Permissions
+
+DataHub provides granular access control for Iceberg operations through policies. The following privileges were introduced with Iceberg support:
+
+| Operation | Required Privilege | Resource Type |
+|-----------|-------------------|---------------|
+| CREATE or DROP namespaces | Manage Namespaces | Data Platform Instance |
+| CREATE, ALTER or DROP tables | Manage Tables | Data Platform Instance |
+| CREATE, ALTER or DROP views | Manage Views | Data Platform Instance |
+| SELECT from tables or views | Read Only data-access | Dataset |
+| INSERT, UPDATE, DELETE or ALTER tables | Read-write data-access | Dataset |
+| List tables or views | List tables, views and namespaces | Data Platform Instance |
+
+To configure access:
+1. Create policies with the appropriate privileges for Iceberg users
+<p>
+  <img width="70%"  src="https://raw.githubusercontent.com/chakru-r/static-assets/b569b7bc6805dd459640cdf44649eb2c75d8c846/imgs/iceberg-policy-type.png"/>
+</p>
+
+2. Scope the policies by:
+    - Selecting specific warehouse Data Platform Instances for namespace, table management, and listing privileges
+    - Selecting specific Datasets for table and view data access privileges
+<p>
+  <img width="70%"  src="https://raw.githubusercontent.com/chakru-r/static-assets/b569b7bc6805dd459640cdf44649eb2c75d8c846/imgs/iceberg-policy-privileges.png"/>
+</p>
+
+3. Assign the policies to relevant users or groups
+<p>
+  <img width="70%"  src="https://raw.githubusercontent.com/chakru-r/static-assets/b569b7bc6805dd459640cdf44649eb2c75d8c846/imgs/iceberg-policy-users-groups.png"/>
+</p>
+
```

Comment:
Also add screenshots for what people will see after their job runs in terms of the dataset entity on DataHub